docs/make.jl

```diff
@@ -16,53 +16,103 @@ if !(haskey(ENV, "GITHUB_ACTIONS") || haskey(ENV, "GITLAB_CI"))
   config["repo_root_url"] = "https://github.com/AlgebraicJulia/Decapodes.jl/blob/main/docs"
 end
 
-# const literate_dir = joinpath(@__DIR__, "..", "examples")
-# const generated_dir = joinpath(@__DIR__, "src", "examples")
+const literate_dir = joinpath(@__DIR__, "..", "examples")
+const generated_dir = joinpath(@__DIR__, "src", "examples")
+
+@info "Building literate files"
+for (root, dirs, files) in walkdir(literate_dir)
+  out_dir = joinpath(generated_dir, relpath(root, literate_dir))
+  pmap(files) do file
+    f,l = splitext(file)
+    if l == ".jl" && !startswith(f, "_")
+      Literate.markdown(joinpath(root, file), out_dir;
+        config=config, documenter=true, credit=false)
+      Literate.notebook(joinpath(root, file), out_dir;
+        execute=true, documenter=true, credit=false)
+    end
+  end
+end
+@info "Completed literate"
+
+pages = Any[]
+push!(pages, "Decapodes.jl"      => "index.md")
+push!(pages, "Overview"          => "overview.md")
+push!(pages, "Equations"         => "equations.md")
+push!(pages, "BC Debug"          => "bc_debug.md")
+push!(pages, "ASCII Operators"   => "ascii.md")
```

Comment:
since this part is being hard coded, it could still be a `Vector{Pair{String,String}}` literal and then you can push to it for the dynamically generated part later.